lib/ansible/modules/network/eos/eos_interfaces.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+##############################################
+#                 WARNING                    #
+##############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+##############################################
+
+"""
+The module file for eos_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = """
+---
+module: eos_interfaces
+version_added: 2.9
+short_description: Manages interface attributes of Arista EOS interfaces
+description: ['This module manages the interface attributes of Arista EOS interfaces.']
+author: ['Nathaniel Case (@qalthos)']
+options:
+  config:
+    description: The provided configuration
+    type: list
+    suboptions:
+      description:
+        description:
+        - Interface description
+        type: str
+      duplex:
+        description:
+        - Interface link status. Applicable for Ethernet interfaces only.
+        - Values other than C(auto) must also set I(speed).
+        - Ignored when I(speed) is set above C(1000).
+        type: str
+      enabled:
+        default: true
+        description:
+        - Administrative state of the interface.
+        - Set the value to C(true) to administratively enable the interface or C(false)
+          to disable it.
+        type: bool
+      mtu:
+        description:
+        - MTU for a specific interface. Must be an even number between 576 and 9216.
+          Applicable for Ethernet interfaces only.
+        type: int
+      name:
+        description:
+        - Full name of the interface, e.g. GigabitEthernet1.
+        type: str
```

Comment:
If we can move `name` at top of subotion list that would be in sync with other platform, also coz it's the only `required` param for this resource module.